File: src/main/java/com/andrew121410/mc/ccminecraftbot/config/ConfigUtils.java
package com.andrew121410.mc.ccminecraftbot.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
import lombok.SneakyThrows;

import java.io.File;

public class ConfigUtils {

    @SneakyThrows
    public static CCMinecraftBotJacksonConfig loadMainConfig() {
        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
        File configFolder = new File("config");
        File configFile = new File(configFolder, "config.yml");
        if (!configFolder.exists()) {
            configFolder.mkdir();
            objectMapper.writeValue(configFile, new CCMinecraftBotJacksonConfig());
            return null;
        }
        return objectMapper.readValue(configFile, CCMinecraftBotJacksonConfig.class);
    }
}

File: src/main/kotlin/com/andrew121410/mc/ccminecraftbot/config/ConfigUtils.kt
package com.andrew121410.mc.ccminecraftbot.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import java.io.File

object ConfigUtils {
    @Throws
    fun loadConfig(): CCMinecraftBotJacksonConfig? {
        val objectMapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
        val configFolder = File("config")
        val configFile = File(configFolder, "config.yml")
        if (!configFolder.exists()) {
            configFolder.mkdir()
            objectMapper.writeValue(configFile, CCMinecraftBotJacksonConfig())
            return null
        }
        return objectMapper.readValue(configFile, CCMinecraftBotJacksonConfig::class.java)
    }
}